\documentclass[12pt,a4paper]{book}
\usepackage[]{graphicx}
\usepackage[]{color}
%% maxwidth is the original width if it is less than linewidth
%% otherwise use linewidth (to make sure the graphics do not exceed the margin)
\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\definecolor{fgcolor}{rgb}{0.345, 0.345, 0.345}
\newcommand{\hlnum}[1]{\textcolor[rgb]{0.686,0.059,0.569}{#1}}%
\newcommand{\hlstr}[1]{\textcolor[rgb]{0.192,0.494,0.8}{#1}}%
\newcommand{\hlcom}[1]{\textcolor[rgb]{0.678,0.584,0.686}{\textit{#1}}}%
\newcommand{\hlopt}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlstd}[1]{\textcolor[rgb]{0.345,0.345,0.345}{#1}}%
\newcommand{\hlkwa}[1]{\textcolor[rgb]{0.161,0.373,0.58}{\textbf{#1}}}%
\newcommand{\hlkwb}[1]{\textcolor[rgb]{0.69,0.353,0.396}{#1}}%
\newcommand{\hlkwc}[1]{\textcolor[rgb]{0.333,0.667,0.333}{#1}}%
\newcommand{\hlkwd}[1]{\textcolor[rgb]{0.737,0.353,0.396}{\textbf{#1}}}%

\usepackage{framed}
\makeatletter
\newenvironment{kframe}{%
 \def\at@end@of@kframe{}%
 \ifinner\ifhmode%
  \def\at@end@of@kframe{\end{minipage}}%
  \begin{minipage}{\columnwidth}%
 \fi\fi%
 \def\FrameCommand##1{\hskip\@totalleftmargin \hskip-\fboxsep
 \colorbox{shadecolor}{##1}\hskip-\fboxsep
     % There is no \\@totalrightmargin, so:
     \hskip-\linewidth \hskip-\@totalleftmargin \hskip\columnwidth}%
 \MakeFramed {\advance\hsize-\width
   \@totalleftmargin\z@ \linewidth\hsize
   \@setminipage}}%
 {\par\unskip\endMakeFramed%
 \at@end@of@kframe}
\makeatother

\definecolor{shadecolor}{rgb}{.97, .97, .97}
\definecolor{messagecolor}{rgb}{0, 0, 0}
\definecolor{warningcolor}{rgb}{1, 0, 1}
\definecolor{errorcolor}{rgb}{1, 0, 0}
\newenvironment{knitrout}{}{} % an empty environment to be redefined in TeX

\usepackage{alltt}

\usepackage{geometry}
\usepackage{graphicx}
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\begin{document}
%\title{A short material for \textit{Introduction to Computational Biology}}
%\author{Songpeng Zu\\Department of Automation, Tsinghua University}
%\date{\today}
%\maketitle
\begin{titlepage}
        \centering
%	\includegraphics[width=0.15\textwidth]{example-image-1x1}\par\vspace{1cm}
        {\scshape\LARGE Tsinghua University \par}
        \vspace{1cm}
        {\scshape\Large Reference for undergraduate\par}
        \vspace{1.5cm}
        {\huge\bfseries A Short Material for \textit{Introduction to Computational Biology}\par}
        \vspace{2cm}
        {\Large\itshape Songpeng Zu\par}
        \vfill
        supervised by\par
        Dr. Shao Li
%        Dr.~Shao~\textsc{Li}
        \vfill
% Bottom of the page
        {\large \today\par}
\end{titlepage}
\chapter*{Preface}
This material is specific for an undergraduate class called
{\it Introduction to Computational Biology}, which is taught by Dr. Shao Li in
Tsinghua University. In this class, we mainly cover:
\begin{itemize}
\item Basic knowledge about the central law from DNA to protien.
\item Basic diverse omics knowledge and analysis.
\item Network pharmacology and modern traditional Chinese medicine.
\end{itemize}
We also have several exercise classes focus on computational methods on
bioinformatics, such as
\begin{itemize}
\item Sequence alignment, like Smith-Waterman algorihtm, multiple
  sequence alignment.
\item Unsupervised machine learning methods, such as principle component
  analysis, clustering (hierarchical clustering, K-means) for gene expression
  data, phenotype-genotype data or other related omics data.
\item Supervised machine learning methods, such as linear regression, logistic
  regression, LASSO for predicting drug responses based on genomic data and so on.
\end{itemize}
In other words, this class involves both basic biological knowledge and basic
machine learning tools for bioinformatics. \\
Since bioinformatics is a extremely hot field, and lots of papers are being published
to change our traditional ways to see, to analyze the biologcial data. It's
quite important and effective way for students to take active part in our class,
especially read recently published high-quality articles and make the
presentations in class. Reading publised articals can make students quickly
understand lots of concepts, current prograss, and how we do research on
bioinformatics. \\
The difficulty lies on the limited time for our class (usually
one 90-minute class for one week, and in total 16 weeks for one
semester). In the past, we have five classes as the exercise classes to talk
about the frequently used computational methods. And studnets are hard to take
active part in our classes. In this year, we try a new way. Around 24 students
were classified as three groups, and each group focuses on one particular topics
in bioinformatics, such as sequence analysis, genome-wide association study and
network pharamcology. Each group will present one recently publishsed article
for 20 minutes in one exercise class. And we use the left 30 minutes to talk
about the computational methods. Students need to pay more time to teach
themselves these methods to complete the homework, and also prepare for the
paper presentations. \\
It is indeed an interesting try, and we find students are quitely interested in
the class now. The shorcut is that we need a specific materials for our exercise
class since we cover several parts of machine learning, statistics, and
optimization in the limited time.\\
So we decide to write the specific material for our exercise class, and hope it
would be helpful for students to teach themselves.


\end{document}

%%% Local Variables:
%%% mode: latex
%%% TeX-master: t
%%% End:
